__backend_test__/packagesEndpoint.test.ts

```diff
@@ -43,4 +43,140 @@ describe('getPackageHistory', () => {
       ID: data.id,
     })));
   });
+});
+
+//GET package download test
+describe('getPackageDownload', () => {
+  it('should return package download data when valid package name and version are provided', async () => {
+    const req: Partial<Request> = {
+      query: {
+        name: 'underscore',
+        version: '1.2.3',
+      },
+    };
+
+    let responseObject = {};
+
+    const res: Partial<Response> = {
+      status: jest.fn().mockReturnThis(),
+      json: jest.fn().mockImplementation((JSONdata: PackageDownloadResponseType) => {
+        responseObject = JSONdata;
+      }),
+      send: jest.fn().mockReturnThis(),
+    };
+
+    const mockMetaData = {
+      id: 'metadataId',
+      name: 'underscore',
+      version: '1.2.3',
+    };
+
+    const mockData = {
+      id: 'dataId',
+      content: 'mockContent',
+      URL: 'mockURL',
+      JSProgram: 'mockJSProgram',
+    };
+
+    // Mock the database call to return the mock metadata and data
+    (prismaCalls.getPackageWithMetadataAndData as jest.Mock).mockResolvedValue({ metadata: mockMetaData, data: mockData });
+
+    await getPackageDownload(req as Request, res as Response);
+
+    expect(res.status).toHaveBeenCalledWith(200);
+    expect(res.json).toHaveBeenCalledWith({
+      metadata: {
+        Name: mockMetaData.name,
+        Version: mockMetaData.version,
+        ID: mockMetaData.id,
+      },
+      data: {
+        Content: mockData.content,
+        URL: mockData.URL,
+        JSProgram: mockData.JSProgram,
+      },
+    });
+  });
+});
+
+//Update package test
+describe('updatePackage', () => {
+  it('should successfully update the package details when valid data is provided', async () => {
+    const req: Partial<Request> = {
+      body: {
+        name: 'underscore',
+        version: '1.2.3',
+        content: 'updatedContent',
+        URL: 'updatedURL',
+        JSProgram: 'updatedJSProgram',
+      } as PackageUpdateRequestType,
+    };
```

Comment:
{
  "metadata": {
    "Name": "string",
    "Version": "1.2.3",
    "ID": "string"
  },
  "data": {
    "Content": "string",
    "URL": "string",
    "JSProgram": "string"
  }
}

This is how the request will look like. So your example request is wrong.